util/ranger/points.go

```diff
@@ -250,6 +250,12 @@ func (r *builder) buildFormBinOp(expr *expression.ScalarFunction) []*point {
 			// If the original value is adjusted, we need to change the condition.
 			// For example, col < 2156. Since the max year is 2155, 2156 is changed to 2155.
 			// col < 2155 is wrong. It should be col <= 2155.
+
+			// If nulleq with null value, values.ToInt64 will return err
+			if value.IsNull() {
```

Comment:
Nitpick, I prefer to put this check into line 249, i.e, `col.GetType().Tp == mysql.TypeYear && !value.IsNull()`.